### DISCUSSION TITLE
Windows 7 Integration Patch Exchange

### DISCUSSION THREAD

**Discussion by WYAYZM** [WYAYZM/Hobby-research-community#8] [General]
Come share your experiences

**Comment by WYAYZM** [comment]
# Windows 7打补丁关键性版本变更补丁记录

C:\Users\Administrator>DISM.exe /Image:C:\test\offline /Add-Package /PackagePath:C:\test\packages\Windows6.1-KB3020369-x64.cab

部署映像服务和管理工具
版本: 10.0.19041.844

映像版本: 6.1.7601.17514

正在处理 1 (共 1) - 正在添加程序包 Package_for_KB3020369~31bf3856ad364e35~amd64~~6.1.1.1
[==========================100.0%==========================]
操作成功完成。

C:\Users\Administrator>DISM.exe /Image:C:\test\offline /Add-Package /PackagePath:C:\test\packages\Windows6.1-KB3020370-x64.cab

部署映像服务和管理工具
版本: 10.0.19041.844

映像版本: 6.1.7601.18766

正在处理 1 (共 1) - 正在添加程序包 Package_for_KB3020370~31bf3856ad364e35~amd64~~6.1.1.2
[==========================100.0%==========================]
操作成功完成。

# Windows 7打补丁关键性版本变更补丁记录2

C:\Users\Administrator>DISM.exe /Image:C:\test\offline /Add-Package /PackagePath:C:\test\packages\Windows6.1-KB4490628-x64.cab

部署映像服务和管理工具
版本: 10.0.19041.844

映像版本: 6.1.7601.18766

正在处理 1 (共 1) - 正在添加程序包 Package_for_KB4490628~31bf3856ad364e35~amd64~~6.1.1.2
[==========================100.0%==========================]
操作成功完成。

C:\Users\Administrator>DISM.exe /Image:C:\test\offline /Add-Package /PackagePath:C:\test\packages\Windows6.1-KB4525251-x64.cab

部署映像服务和管理工具
版本: 10.0.19041.844

映像版本: 6.1.7601.24383

正在处理 1 (共 1) - 正在添加程序包 Package_for_RollupFix~31bf3856ad364e35~amd64~~7601.24536.1.11
[==========================100.0%==========================]
操作成功完成。

# Windows 7打补丁关键性版本变更补丁记录3

C:\Users\Administrator>DISM.exe /Image:C:\test\offline /Add-Package /PackagePath:C:\test\packages\Windows6.1-KB4531786-x64.cab

部署映像服务和管理工具
版本: 10.0.19041.844

映像版本: 6.1.7601.24383

正在处理 1 (共 1) - 正在添加程序包 Package_for_KB4531786~31bf3856ad364e35~amd64~~6.1.1.0
[==========================100.0%==========================]
操作成功完成。

C:\Users\Administrator>DISM.exe /Image:C:\test\offline /Add-Package /PackagePath:C:\test\packages\Windows6.1-KB4532945-x64.cab

部署映像服务和管理工具
版本: 10.0.19041.844

映像版本: 6.1.7601.24537

正在处理 1 (共 1) - 正在添加程序包 Package_for_KB4532945~31bf3856ad364e35~amd64~~6.1.1.1
[==========================100.0%==========================]
操作成功完成。

# Windows 7打补丁关键性版本变更补丁记录4

C:\Users\Administrator>DISM.exe /Image:C:\test\offline /Add-Package /PackagePath:C:\test\packages\Windows6.1-KB5017397-x64.cab

部署映像服务和管理工具
版本: 10.0.19041.844

映像版本: 6.1.7601.24537

正在处理 1 (共 1) - 正在添加程序包 Package_for_KB5017397~31bf3856ad364e35~amd64~~6.1.1.0
[==========================100.0%==========================]
操作成功完成。

C:\Users\Administrator>DISM.exe /Image:C:\test\offline /Add-Package /PackagePath:C:\test\packages\Windows6.1-KB5020000-x64.cab

部署映像服务和管理工具
版本: 10.0.19041.844

映像版本: 6.1.7601.26111

正在处理 1 (共 1) - 正在添加程序包 Package_for_RollupFix~31bf3856ad364e35~amd64~~7601.26221.1.11
[==========================100.0%========================= ]
发生错误 - Package_for_RollupFix 错误: 0x80073713

错误: 14099

指定的映像不再可用。
请卸载该映像并放弃更改。然后装载原始映像以重试该操作。

可以在 C:\Windows\Logs\DISM\dism.log 上找到 DISM 日志文件

**Comment by WYAYZM** [comment]
Windows 7的补丁列表整理真的是难搞哦

**Comment by WYAYZM** [comment]
# Windows 7 SP1 x64 MSDN下载的版本已安装系统映像程序包列表

程序包列表:

程序包标识 : Microsoft-Windows-Client-Refresh-LanguagePack-Package~31bf3856ad364e35~amd64~zh-CN~6.1.7601.17514
状态 : 安装挂起
发行类型 : Language Pack
安装时间 : 2011/4/12/周二 14:57

程序包标识 : Microsoft-Windows-CodecPack-Basic-Package~31bf3856ad364e35~amd64~~6.1.7601.17514
状态 : 已安装
发行类型 : Feature Pack
安装时间 : 2010/11/21/周日 3:40

程序包标识 : Microsoft-Windows-Foundation-Package~31bf3856ad364e35~amd64~~6.1.7601.17514
状态 : 已安装
发行类型 : Foundation
安装时间 : 2010/11/21/周日 3:40

程序包标识 : Microsoft-Windows-IE-Troubleshooters-Package~31bf3856ad364e35~amd64~en-US~6.1.7601.17514
状态 : 卸载挂起
发行类型 : Language Pack
安装时间 : 2010/11/21/周日 3:40

程序包标识 : Microsoft-Windows-IE-Troubleshooters-Package~31bf3856ad364e35~amd64~~6.1.7601.17514
状态 : 已安装
发行类型 : Feature Pack
安装时间 : 2010/11/21/周日 3:40

程序包标识 : Microsoft-Windows-InternetExplorer-Optional-Package~31bf3856ad364e35~amd64~en-US~8.0.7601.17514
状态 : 卸载挂起
发行类型 : Language Pack
安装时间 : 2010/11/21/周日 3:40

程序包标识 : Microsoft-Windows-InternetExplorer-Optional-Package~31bf3856ad364e35~amd64~~8.0.7601.17514
状态 : 已安装
发行类型 : Feature Pack
安装时间 : 2010/11/21/周日 3:40

程序包标识 : Microsoft-Windows-LocalPack-CN-Package~31bf3856ad364e35~amd64~~6.1.7601.17514
状态 : 安装挂起
发行类型 : Local Pack
安装时间 : 2011/4/12/周二 15:00

程序包标识 : Package_for_KB2534111~31bf3856ad364e35~amd64~~6.1.1.0
状态 : 安装挂起
发行类型 : Hotfix
安装时间 : 2011/4/12/周二 14:47

程序包标识 : Package_for_KB976902~31bf3856ad364e35~amd64~~6.1.1.17514
状态 : 已安装
发行类型 : Update
安装时间 : 2010/11/21/周日 3:01

**Reply by WYAYZM** [reply]
发行类型 : Hotfix，可能又不懂的人想真的这个Hotfix发行类型是什么，其实这个发行类型的意思是热修补，也就是热修补补丁，即online，在系统安装修复的

**Reply by windowslove** [reply]
...

**Reply by WYAYZM** [reply]
> ...

咋了？

**Reply by WYAYZM** [reply]
> ...

有什么问题吗

**Reply by WYAYZM** [reply]
请说出你的疑问

**Comment by WYAYZM** [comment]
KB3042058这个前147补丁里需要剔除出来单独安装才能成功

**Reply by windowslove** [reply]
没啥疑问

**Reply by WYAYZM** [reply]
哈哈

**Reply by WYAYZM** [reply]
期待我的成品出炉，争取在10号前做出来
